### ISSUE TITLE
Refresh Discord Rules/FAQ/Pins and the Beginning of Foundations

### ISSUE THREAD

**Issue by rlmoser99** [TheOdinProject/top-meta#66]
| Title | Author | Date |
| --- | --- | --- |
| Refresh Rules, FAQ, Pinned Posts, and Beginning of Foundations | Rachel | Oct 2021 |

# Refresh Discord Rules/FAQ/Pins and the Beginning of Foundations

## Summary

As we see a need, we add rules, FAQs, and lengthy pinned posts to Discord, as well as adding content to the beginning of Foundations. Each of these additions are necessary, but over time I think things may be getting a bit disorganized. I'd like to re-evaluate the location of our rules, lengthy pinned posts, and lengthy FAQ answers. Plus, take a fresh look at the beginning of Foundations to make sure that its content is organized into smaller, focused lessons like our newer content.

## Motivation

I want to add non-Discord specific bot commands, pinned posts, FAQ to appropriate places throughout the curriculum. so that everyone has a chance to know all of the helpful information that we frequently utilize in Discord. In addition, using links to previous Discord posts do not consistently take you to the right post, so I'd like to move these regularly referenced pinned posts outside of Discord.

## Suggested implementation

The scope of this proposal can be really broad, but here are a few specific things that are on my mind that I'd like to see updated.

1. The "Join The Odin Community" lesson is really long and it covers similar information as the "Asking For Help" lesson. Plus, I'd like to break up the "Join the Odin Community" lesson into smaller lessons that make it easier to reference.

2. We have discussed revising a few rules lately but I do not think that anyone followed through making the changes. Updating our rules on Discord is a bit complicated, plus we need to keep it under the 150 character limit for the new screening process. I think it might be time to consider putting the rules on our website, and just having abbreviated versions in the screening process.

3. Recently I read a critique that we do not discuss operating systems in the initial lesson, How This Course Will Work, since it is such a vital step of how this course will work. I'd like to introduce our OS requirements & some of our reasoning to that lesson, because I believe that it should be covered in this lesson.

4. Determine which bot commands, faq answers, and pinned posts are not specific to Discord and valuable to all of our students. Add them to a relevant place in the curriculum.

5. Determine which of our popular & lengthy pinned posts should be relocated. I'd like to make them open source on our website, as opposed to a blog somewhere else, so that they can be easily revised in the future.

## Drawbacks

This is additional work, without any significant new material.

## Alternatives

We could just leave everything the way it is now.

## Additional

There are many little decisions that need to be discussed to help direct this specifics of this project. As a starting point, here are some questions to get the ball rolling:

* Should we add a "Join Our Odin Community" section to Foundations to extract smaller lessons out of the really long "Join the Odin Community" lesson?
* An alternative to the above would be to create a new page outside of Foundations that contains all of our "Join our Discord Community" material? We could then have all of our current Discord links point to this new page (and also ensure that anyone that joins Discord from our website sees this material).
* Should we move the Discord rules to our website or is there another way to make them more accessible for future changes?
* If we move our Discord rules to our website, should we consider moving some of the lengthier Discord FAQ answers & popular pins?
* Where is the best place for our lengthy FAQ answers and/or popular pinned posts - a blog or do we want to keep them open source (and easily updated)?
* How do we want to reword the Dogpiling rule?
* How do we want to reword rules 11, 12, 13 since they are often used together?
* Any other rules changes that have been discussed recently?
* Is anyone opposed to adding OS to our very first lesson, How This Course Will Work?

**Comment by twalton83** [comment]
I am +1 for moving to the website and reassessing possibly expanding rules into a different format based on community interactions of late. 

Examples: [Rust Community Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct)
[Reactiflux Community Code of Conduct](https://www.reactiflux.com/conduct)

These are comprehensive codes of conduct that bring transparency to moderation and community expectations, which may be a bit lacking here. This also can't all fit nicely on Discord.

**Comment by rlmoser99** [comment]
@twalton83 Right now I am working on moving the community things out of the Foundations lessons, so that no matter when someone joins Discord, they are aware of our expected etiquette & also how to ask a question. I do not have the PR done yet, but I am making 3 pages. 

1. Community Expectations - which is currently in this [assignment](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/join-the-odin-community#assignment).
2. Help Yourself Before Asking Others - which started out as this [section](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/join-the-odin-community#before-asking-for-help).
3. How to Ask Technical Questions (which is a combination of this [section](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/join-the-odin-community#asking-for-help), this [lesson](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/asking-for-help) and bits of /question article.  

The basic idea will be that a user will be given [steps](https://tailwindui.com/components/application-ui/navigation/steps) on their dashboard to complete before they are given the link to join Discord. I hope to have some visuals to share at our meeting in 2 weeks. I can easily make the rules one of these steps, or combine it with the community expectations page. 

As you can tell, this will be quite a big change to a user's workflow to join Discord, if the team decides to go with it. My only concern is that I am making these steps longer than a user will actually take the time to read. However, in the end I think it will improve the quality of questions from people joining Discord in places outside of that one spot in Foundations & creating new pages on our website that will be really easy to link people to when they need to re-read it.

**Comment by twalton83** [comment]
I agree. I think that if this is good enough for other communities, it’s good enough for us, and we simply need well documented code of conduct and expectations to point to for moderation and when setting boundaries in the chat. 

While I agree with the concern about folks not reading, that is not on us and is an expectation, and violating those expectations is an issue of moderation.
